server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -2966,9 +2966,20 @@ public UserVm destroyVm(DestroyVMCmd cmd) throws ResourceUnavailableException, C
         checkForUnattachedVolumes(vmId, volumesToBeDeleted);
         validateVolumes(volumesToBeDeleted);
 
+        List<VolumeVO> allVolumes = null;
+        if (vm.getHypervisorType() == HypervisorType.VMware) {
+            allVolumes = _volsDao.findByInstance(vm.getId());
+            allVolumes.removeIf(vol -> vol.getVolumeType() == Volume.Type.ROOT);
+        }else {
+            allVolumes = volumesToBeDeleted;
+        }
```

Comment:
@lujiefsi I believe this section isn't required as it causes issues wrt recovery of destroyed VMs with their data disks on VMWare (https://github.com/apache/cloudstack/pull/4493/files#diff-865f710fe60b9b62ff8c2cbce8a8342e4f6fca658223119762285fd5f7857f08L2949).

Reply:
ops, have fixed it